contribs/sharing/src/main/java/org/matsim/contrib/sharing/run/SharingQSimServiceModule.java

```diff
@@ -58,6 +59,17 @@ protected void configureQSim() {
 					specification.getVehicles(), network, serviceConfig.getMaximumAccessEgressDistance());
 		})).in(Singleton.class);
 
+
+		String agentSource = SharingUtils.getServiceMode(serviceConfig)+AGENT_SOURCE_SUFFIX;
+		addQSimComponentBinding(agentSource).toProvider(modalProvider(getter -> {
```

Comment:
Here you can use addModalComponent to avoid the "named component" and having AGENT_SOURCE_SUFFIX.

Reply:
Changed to addModalComponent